events.go

```diff
@@ -0,0 +1,133 @@
+package main
+
+import (
+	"github.com/veandco/go-sdl2/sdl"
+	"time"
+)
+
+type SdlEventStream struct {
+	closing chan chan error
+	events  chan sdl.Event
+}
+
+func (ses *SdlEventStream) Close() error {
+	c := make(chan error)
+	ses.closing <- c
+	return <-c
+}
+
+func (ses *SdlEventStream) loop() {
+	var err error
+	for {
+		evt := sdl.PollEvent()
+		select {
+		case errc := <-ses.closing:
+			errc <- err
+			close(ses.events)
+			return
+		case <-time.After(1 * time.Second):
+		default:
+			if evt != nil {
+				ses.events <- evt
+			}
+		}
+	}
+}
+
+// Receive returns a read only channel where sdl Events will
+// be submitted
+func (ses *SdlEventStream) Receive() <-chan sdl.Event {
+	// This makes it possible to call this more than once
+	// but don't have this starting two loops or more
+	if ses.events == nil {
+		ses.events = make(chan sdl.Event)
+		ses.closing = make(chan chan error)
+		go ses.loop()
+	}
```

Comment:
Init loop and channel in New method in order to avoid racy usage (RAII-Pattern). The zero value is not really usable for SdlEventStream.